.travis.yml

```diff
@@ -2,7 +2,7 @@ services:
   - docker
 
 git:
-  depth: 1
+  depth: 2
```

Comment:
Can you explain a little why a `git.depth` of 2 is used?
Is it sufficient in all cases? (ex: if a PR contains multiple commits when appropriate).

Reply:
A depth of 2 ensures that the last two commits to a specific branch are available. 
The check_changes script then retrieves the changes between these 2 commits and filters. Only changes for a certain directory tree are returned:

`git diff --name-status $TRAVIS_BRANCH~1...$TRAVIS_BRANCH .`

So every time a new commit is pushed to a pull request and it contains changes to files inside the "$IMAGE_NAME-docker" folder then the image will be rebuilt.

Reply:
No need to change the depth anymore , a depth of 1 is fine since we do a diff between HEAD and HEAD~1

Reply:
Ok, will update it

Reply:
Hmm, I did a quick test. --depth 2 is required:

```
git clone --depth 1 git@github.com:theia-ide/theia-apps.git
cd theia-apps
$ git diff HEAD HEAD~1
fatal: ambiguous argument 'HEAD~1': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

With "-depth 2" this command works as expected. A depth of one ensures that the HEAD is available, a depth of 2 also ensures that we can compare with the previous commit.

Reply:
I tried your code plus 3 commits (different apps) on top of it ( https://travis-ci.org/theia-ide/theia-apps/builds/586164681 ) and I put depth = 1.
Everything seems fine. Can you give me an example when I need depth=2
Thanks

Reply:
I think `depth=1` is enough - the way Travis works you'll still have all you need. See this example: https://travis-ci.org/theia-ide/theia-apps/jobs/584695052#L444-L455

That being said, the extra resources used with `depth=2` are negligible.

Reply:
Ok, reverted